x-pack/plugins/spaces/server/new_platform/spaces_service/spaces_service.ts

```diff
@@ -0,0 +1,138 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import { first } from 'rxjs/operators';
+import { Observable, Subscription } from 'rxjs';
+import { Legacy } from 'kibana';
+import { SavedObjectsService } from 'src/legacy/server/kbn_server';
+import {
+  Logger,
+  ElasticsearchServiceSetup,
+  HttpServiceSetup,
+  KibanaRequest,
+} from 'src/core/server';
+import { DEFAULT_SPACE_ID } from '../../../common/constants';
+import { SecurityPlugin } from '../../../../security';
+import { SpacesClient } from '../../lib/spaces_client';
+import { getSpaceIdFromPath } from '../../lib/spaces_url_parser';
+import { SpacesConfig } from '../config';
+
+type RequestFacade = KibanaRequest | Legacy.Request;
+
+export interface SpacesServiceSetup {
+  scopedClient(request: RequestFacade): SpacesClient;
+
+  getSpaceId(request: RequestFacade): string;
+
+  isInDefaultSpace(request: RequestFacade): boolean;
+}
+
+interface CacheEntry {
+  spaceId: string;
+  isInDefaultSpace: boolean;
+}
+
+interface SpacesServiceDeps {
+  http: HttpServiceSetup;
+  elasticsearch: ElasticsearchServiceSetup;
+  savedObjects: SavedObjectsService;
+  getSecurity: () => SecurityPlugin | undefined;
+  config$: Observable<SpacesConfig>;
+  spacesAuditLogger: any;
+}
+
+export class SpacesService {
+  private configSubscription$?: Subscription;
+
+  private readonly contextCache: WeakMap<any, CacheEntry> = new WeakMap();
```

Comment:
do we really need another cache? `getBasePathFor` uses cache internally and should handle all cases when url has changed

Reply:
potentially not -- the original goal was to prevent re-calculation of the request's space identifier. In reality, it's likely not a performance issue. I _think_ this used to be an `async` call that required a round-trip to ES. That's not the case anymore, so the benefits of the cache are somewhat diminished. 

@kobelb did I miss something? Do you agree the cache is likely unnecessary?

Reply:
> @kobelb did I miss something? Do you agree the cache is likely unnecessary?

I don't see anything that you missed, I agree that removing the cache should be fine and we should only take a small performance hit.